docs/modules/ROOT/pages/contributors/release.adoc

```diff
@@ -0,0 +1,104 @@
+= Release Process
+
+This document explains the requirements for and the process of a KubeArchive release.
+
+== Commits
+
+The KubeArchive release process scans commits from the latest tag up to the current
+commit. The commits must mention the GitHub Pull Requests they come from on their
+title (first line). A commit merged from the Pull Request with ID 100 should have
+this title:
+
+[source,text]
+----
+Add feature A (#100)
+----
+
+The Pull Request mention is added by default when the commits are squashed on
+Pull Request merge.
+
+== Pull Request Labels
+
+KubeArchive follows the
+link:https://semver.org/[Semantic Versioning]
+system with a leading `v`: `vMAJOR.MINOR.PATCH`. To summarize, `MAJOR` releases
+break user-facing interfaces, `MINOR` releases introduce new features and `PATCH`
+versions introduce bug fixes or documentation changes.
+
+The KubeArchive release process relies on the labels on the GitHub Pull Requests
+when determining the next version number for the release. The following labels
+impact which number is updated in the next release:
+
+* `kind/breaking`: next release is a MAJOR version. MINOR and PATCH numbers are reset to 0.
+* `kind/feature`: if `kind/breaking` is not present, next release is a MINOR version.
+PATCH number is reset to 0.
+* `kind/documentation` and `kind/bug`: if `kind/breaking` and `kind/feature` are
+not present, next release is a PATCH version.
+
+A Pull Request requires at least one of these labels.
+
+== Release Notes
+
+The KubeArchive release process creates release notes for users. These notes are
+generated from the GitHub Pull Requests that contain a code block annotated with
+`release-note`:
+
+[source,text]
+----
+```release-note
+Introduced new feature called `feature-a`. To enable it, set to `true` the
+`feature-a` flag in the `feature-flags` ConfigMap in the `kubearchive` namespace.
+```
+----
+
+To not include release notes from the Pull Request the release-note block must
+contain the string `NONE`:
+
+[source,text]
+----
+```release-note
+NONE
+```
+----
```

Comment:
Can we add an example of when this is expected to happen? I guess that adding tests would be an example of not needing release notes in a PR, right?

Reply:
I added some examples in the previous paragraphs. Let me know what you think